test/linalg/qr.jl

```diff
@@ -38,107 +37,108 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
         @test abs(qrfact(α)[:Q][1,1]) ≈ one(eltyb)
         tab = promote_type(eltya,eltyb)
 
-debug && println("\ntype of a: ", eltya, " type of b: ", eltyb, "\n")
-debug && println("QR decomposition (without pivoting)")
         for i = 1:2
             let a = i == 1 ? a : view(a, 1:n - 1, 1:n - 1), b = i == 1 ? b : view(b, 1:n - 1), n = i == 1 ? n : n - 1
-                qra   = @inferred qrfact(a)
-                @inferred qr(a)
-                q, r  = qra[:Q], qra[:R]
-                @test_throws KeyError qra[:Z]
-                @test q'*full(q, thin=false) ≈ eye(n)
-                @test q*full(q, thin=false)' ≈ eye(n)
-                @test q'*eye(n)' ≈ full(q, thin=false)'
-                @test full(q, thin=false)'q ≈ eye(n)
-                @test eye(n)'q' ≈ full(q, thin=false)'
-                @test q*r ≈ a
-                @test_approx_eq_eps a*(qra\b) b 3000ε
-                @test full(qra) ≈ a
-                @test_approx_eq_eps A_mul_Bc(eye(eltyb,size(q.factors,2)),q)*full(q, thin=false) eye(n) 5000ε
-                if eltya != Int
-                    @test eye(eltyb,n)*q ≈ convert(AbstractMatrix{tab},q)
-                    ac = copy(a)
-                    @test qrfact!(a[:, 1:5])\b == qrfact!(view(ac, :, 1:5))\b
+                @testset "QR decomposition (without pivoting)" begin
+                    qra   = @inferred qrfact(a)
+                    @inferred qr(a)
+                    q, r  = qra[:Q], qra[:R]
+                    @test_throws KeyError qra[:Z]
+                    @test q'*full(q, thin=false) ≈ eye(n)
+                    @test q*full(q, thin=false)' ≈ eye(n)
+                    @test q'*eye(n)' ≈ full(q, thin=false)'
+                    @test full(q, thin=false)'q ≈ eye(n)
+                    @test eye(n)'q' ≈ full(q, thin=false)'
+                    @test q*r ≈ a
+                    @test_approx_eq_eps a*(qra\b) b 3000ε
+                    @test Array(qra) ≈ a
```

Comment:
Re. `Array(qra)`, please see the comment above regarding `full` translating to `AbstractArray` for factorizations.